/* AUTOGENERATED FILE. DO NOT EDIT. */

/*=======Automagically Detected Files To Include=====*/
#include "unity.h"

int GlobalExpectCount;
int GlobalVerifyOrder;
char* GlobalOrderError;

/*=======External Functions This Runner Calls=====*/
extern void setUp(void);
extern void tearDown(void);
extern void test_BasicTryDoesNothingIfNoThrow(void);
extern void test_BasicThrowAndCatch(void);
extern void test_BasicThrowAndCatch_WithMiniSyntax(void);
extern void test_VerifyVolatilesSurviveThrowAndCatch(void);
extern void test_ThrowFromASubFunctionAndCatchInRootFunc(void);
extern void test_ThrowAndCatchFromASubFunctionAndRethrowToCatchInRootFunc(void);
extern void test_ThrowAndCatchFromASubFunctionAndNoRethrowToCatchInRootFunc(void);
extern void test_ThrowAnErrorThenEnterATryBlockFromWithinCatch_VerifyThisDoesntCorruptExceptionId(void);
extern void test_ThrowAnErrorThenEnterATryBlockFromWithinCatch_VerifyThatEachExceptionIdIndependent(void);
extern void test_CanHaveMultipleTryBlocksInASingleFunction(void);
extern void test_CanHaveNestedTryBlocksInASingleFunction_ThrowInside(void);
extern void test_CanHaveNestedTryBlocksInASingleFunction_ThrowOutside(void);
extern void test_AThrowWithoutATryCatchWillUseDefaultHandlerIfSpecified(void);
extern void test_AThrowWithoutOutsideATryCatchWillUseDefaultHandlerEvenAfterTryCatch(void);
extern void test_AbilityToExitTryWithoutThrowingAnError(void);
extern void test_AbilityToExitTryWillOnlyExitOneLevel(void);


/*=======Mock Management=====*/
static void CMock_Init(void)
{
  GlobalExpectCount = 0;
  GlobalVerifyOrder = 0;
  GlobalOrderError = NULL;
}
static void CMock_Verify(void)
{
}
static void CMock_Destroy(void)
{
}

/*=======Test Reset Options=====*/
void resetTest(void);
void resetTest(void)
{
  tearDown();
  CMock_Verify();
  CMock_Destroy();
  CMock_Init();
  setUp();
}
void verifyTest(void);
void verifyTest(void)
{
  CMock_Verify();
}

/*=======Test Runner Used To Run Each Test=====*/
static void run_test(UnityTestFunction func, const char* name, int line_num)
{
    Unity.CurrentTestName = name;
    Unity.CurrentTestLineNumber = line_num;
#ifdef UNITY_USE_COMMAND_LINE_ARGS
    if (!UnityTestMatches())
        return;
#endif
    Unity.NumberOfTests++;
    UNITY_CLR_DETAILS();
    UNITY_EXEC_TIME_START();
    CMock_Init();
    if (TEST_PROTECT())
    {
        setUp();
        func();
    }
    if (TEST_PROTECT())
    {
        tearDown();
        CMock_Verify();
    }
    CMock_Destroy();
    UNITY_EXEC_TIME_STOP();
    UnityConcludeTest();
}

/*=======MAIN=====*/
int main(void)
{
  UnityBegin("test_CExecption.c");
  run_test(test_BasicTryDoesNothingIfNoThrow, "test_BasicTryDoesNothingIfNoThrow", 17);
  run_test(test_BasicThrowAndCatch, "test_BasicThrowAndCatch", 38);
  run_test(test_BasicThrowAndCatch_WithMiniSyntax, "test_BasicThrowAndCatch_WithMiniSyntax", 57);
  run_test(test_VerifyVolatilesSurviveThrowAndCatch, "test_VerifyVolatilesSurviveThrowAndCatch", 77);
  run_test(test_ThrowFromASubFunctionAndCatchInRootFunc, "test_ThrowFromASubFunctionAndCatchInRootFunc", 106);
  run_test(test_ThrowAndCatchFromASubFunctionAndRethrowToCatchInRootFunc, "test_ThrowAndCatchFromASubFunctionAndRethrowToCatchInRootFunc", 149);
  run_test(test_ThrowAndCatchFromASubFunctionAndNoRethrowToCatchInRootFunc, "test_ThrowAndCatchFromASubFunctionAndNoRethrowToCatchInRootFunc", 168);
  run_test(test_ThrowAnErrorThenEnterATryBlockFromWithinCatch_VerifyThisDoesntCorruptExceptionId, "test_ThrowAnErrorThenEnterATryBlockFromWithinCatch_VerifyThisDoesntCorruptExceptionId", 185);
  run_test(test_ThrowAnErrorThenEnterATryBlockFromWithinCatch_VerifyThatEachExceptionIdIndependent, "test_ThrowAnErrorThenEnterATryBlockFromWithinCatch_VerifyThatEachExceptionIdIndependent", 203);
  run_test(test_CanHaveMultipleTryBlocksInASingleFunction, "test_CanHaveMultipleTryBlocksInASingleFunction", 230);
  run_test(test_CanHaveNestedTryBlocksInASingleFunction_ThrowInside, "test_CanHaveNestedTryBlocksInASingleFunction_ThrowInside", 255);
  run_test(test_CanHaveNestedTryBlocksInASingleFunction_ThrowOutside, "test_CanHaveNestedTryBlocksInASingleFunction_ThrowOutside", 282);
  run_test(test_AThrowWithoutATryCatchWillUseDefaultHandlerIfSpecified, "test_AThrowWithoutATryCatchWillUseDefaultHandlerIfSpecified", 309);
  run_test(test_AThrowWithoutOutsideATryCatchWillUseDefaultHandlerEvenAfterTryCatch, "test_AThrowWithoutOutsideATryCatchWillUseDefaultHandlerEvenAfterTryCatch", 321);
  run_test(test_AbilityToExitTryWithoutThrowingAnError, "test_AbilityToExitTryWithoutThrowingAnError", 344);
  run_test(test_AbilityToExitTryWillOnlyExitOneLevel, "test_AbilityToExitTryWillOnlyExitOneLevel", 365);

  return UnityEnd();
}
